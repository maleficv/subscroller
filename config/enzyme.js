import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import fetch from 'isomorphic-fetch';

global.fetch = fetch;

Enzyme.configure({adapter: new Adapter()});