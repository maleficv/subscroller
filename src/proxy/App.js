const http = require('http');
const moment = require('moment');

const getApi = require('./services/getApi');
const logServerEvent = require('./actions/logServerEvent');
const logSubredditSuccess = require('./actions/logSubredditSuccess');
const logSubredditNotFound = require('./actions/logSubredditNotFound');
const saveLogToFile = require('./actions/saveLogToFile');
const loadView = require('./actions/loadView');
const isDev = require('./helpers/isDev');

const port = process.env.PORT || 9005;

const server = http.createServer(async (req, res) => {
    const url = req.url
        .substr(2, req.url.length)
        .split('url=')[1];
        
    const ip = req.headers['x-real-ip'] || req.headers['X-Real-IP'] ||
    req.connection.remoteAddress ||
    req.socket.remoteAddress ||
    (req.connection.socket ? req.connection.socket.remoteAddress : null);

    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');

    if (!url) {
        const view = await loadView('index');
        return res.end(view);
    }
    try {
        const data = await getApi(url);

        !isDev() && logSubredditSuccess(url, ip, moment().format('YYYY/MM/DD hh:mm:ss'), saveLogToFile);

        res.writeHead(200, {'Content-Type': 'application/json'});
        res.end(data);
    } catch (error) {
        !isDev() && logSubredditNotFound(url, moment().format('YYYY/MM/DD hh:mm:ss'), saveLogToFile);

        res.writeHead(404);
        res.end();
    }
});

server.on('error', error => {
    !isDev() && logServerEvent(error, moment().format('YYYY/MM/DD hh:mm:ss'), saveLogToFile);
});

server.listen(port, () => {
    if (isDev()) {
        return console.log(`Server listening on port ${port}`);
    }
    !isDev() && logServerEvent(port, moment().format('YYYY/MM/DD hh:mm:ss'), saveLogToFile);
});

process.on('uncaughtException', () => {
    !isDev() && logServerEvent('process error occurred', moment().format('YYYY/MM/DD hh:mm:ss'), saveLogToFile);
});