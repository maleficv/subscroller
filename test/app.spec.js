import React from 'react';

import {shallow} from 'enzyme';

import App from '../src/app/index';

describe('Root component', () => {
    const app = shallow(<App/>);

    test('App matches snapshot', () => {
        expect(app).toMatchSnapshot();
    });
});