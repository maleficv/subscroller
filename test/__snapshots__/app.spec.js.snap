// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root component App matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Navigation />,
        <Router>
          <Home
            path="/"
            showLightbox={[Function]}
            subreddit="wallpapers"
          />
          <Feed
            path="/subreddit/:subreddit"
            showLightbox={[Function]}
            subreddit="wallpaper"
          />
        </Router>,
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Home
              path="/"
              showLightbox={[Function]}
              subreddit="wallpapers"
            />,
            <Feed
              path="/subreddit/:subreddit"
              showLightbox={[Function]}
              subreddit="wallpaper"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/",
              "showLightbox": [Function],
              "subreddit": "wallpapers",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/subreddit/:subreddit",
              "showLightbox": [Function],
              "subreddit": "wallpaper",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "",
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Navigation />,
          <Router>
            <Home
              path="/"
              showLightbox={[Function]}
              subreddit="wallpapers"
            />
            <Feed
              path="/subreddit/:subreddit"
              showLightbox={[Function]}
              subreddit="wallpaper"
            />
          </Router>,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Home
                path="/"
                showLightbox={[Function]}
                subreddit="wallpapers"
              />,
              <Feed
                path="/subreddit/:subreddit"
                showLightbox={[Function]}
                subreddit="wallpaper"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/",
                "showLightbox": [Function],
                "subreddit": "wallpapers",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/subreddit/:subreddit",
                "showLightbox": [Function],
                "subreddit": "wallpaper",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "",
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
